components/DownloadButton.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect } from 'react'
+import { AntDesign as Icon } from '@expo/vector-icons'
+import * as FileSystem from 'expo-file-system'
+
+import { useMeditation, useAppDispatch } from '../hooks'
+import { filepaths } from '../redux/meditationSlice'
+import { store } from '../redux/store'
+import { useFiles } from '../hooks/useFiles'
+
+export default function DownloadButton(props: Props) {
+  const id = props.id
+  const meditation = useMeditation(id)
+  const files = useFiles('.mp3')
+  const uri = meditation?.uri || ''
+  const [audioFiles, setAudioFiles] = useState<string[]>([])
+  const [downloaded, setDownloaded] = useState(false)
+  const dispatch = useAppDispatch()
+
+  const filename = (path: string) => {
+    let filename = path.split('/').pop()
+    if (!filename) {
+      return
+    }
+    return filename
+  }
+
+  useEffect(() => {
+    let filepaths = store.getState().meditation.filepaths
```

Comment:
We are going to want to use a selector here to avoid undefined variables. I can commit something to fix this up! But essentially you need to add a selector to `selectors.ts` :) 

I'd call it `selectFilePaths`

It would look like this:

```
export const selectFilePaths = (state: RootState) => state.meditation.filepaths || []
```